elm-format-lib/src/Reporting/Error/Syntax.hs

```diff
@@ -490,9 +491,10 @@ data Escape
 
 data Number
   = NumberEnd
-  | NumberDot Int
+  | NumberDot Int64
   | NumberHexDigit
   | NumberNoLeadingZero
+  deriving Show
```

Comment:
This is fine to keep either way, but I'm just curious if `Show` was added simply for debugging, or if it was necessary for things to compile?

Reply:
This is actually needed for now. `show` is used chuck the errors onto a parsec error message.
Here's an example.

https://github.com/avh4/elm-format/blob/8bbada66239b8c0a7c217d6d80cac3db21e7bbff/elm-format-lib/src/Parse/Literal.hs#L23

I also vaguely recall some test helpers relying on `show` to display error messages.

Reply:
Ah okay, so it's needed here but not in elm-compiler because we're using Strings for errors, instead of elm-compiler's customer error type

Reply:
Yes,
if we ever get to the point where all of the parsers have migrated away from parsec then we'll be able to use Elm compiler-style errors all the way, and we wouldn't need `show` (at least not for this reason). Converting the errors to strings done when compiler-style parsers are used in parsec-style parsers, i.e at the border between the old and new code.

(spoiler: In the refactoring PR that I'm currently working on I'll be presenting an attempt at changing the parsec error type so that the Elm compiler's errors can be included without having to be converted to strings)